primitives/arithmetic/src/per_things.rs

```diff
@@ -492,15 +494,34 @@ where
 {
 	let maximum: N = P::ACCURACY.into();
 	let part_n: N = part.into();
-	let c = rational_mul_correction::<N, P>(x.clone(), part, P::ACCURACY, rounding);
+	let c = rational_mul_correction::<N, P>(x.clone(), part, P::ACCURACY, rounding).unwrap();
 	(x / maximum) * part_n + c
 }
 
+/// Wrapper to make all the default name safe. Checking if &denon is zero should cover denom_n
+/// and denom_upper
+fn rational_mul_correction<N, P>(x: N, numer: P::Inner, denom: P::Inner, rounding: Rounding) -> Option<N>
+where
+	N: MultiplyArg + UniqueSaturatedInto<P::Inner>,
+	P: PerThing,
+	P::Inner: Into<N>,
+{
+	if P::Inner::is_zero(&denom) {
+		return None;
+	}
+	Some(unchecked_rational_mul_correction::<N, P>(x, numer, denom, rounding))
+}
+
 /// Compute the error due to integer division in the expression `x / denom * numer`.
 ///
 /// Take the remainder of `x / denom` and multiply by  `numer / denom`. The result can be added
 /// to `x / denom * numer` for an accurate result.
-fn rational_mul_correction<N, P>(x: N, numer: P::Inner, denom: P::Inner, rounding: Rounding) -> N
+fn unchecked_rational_mul_correction<N, P>(
```

Comment:
Existing Rust idioms have `checked_` variants return prefixing, not unchecked variants.

Reply:
super interesting, I was discussing with @ggwpez over chat the rational behind the checked and why functions do not behave as checked ones by default.

 this clarifies everything